src/pydap/model.py

```diff
@@ -287,18 +375,15 @@ class StructureType(DapType, Mapping):
     """A dict-like object holding other variables."""
 
     def __init__(self, name, attributes=None, **kwargs):
-        DapType.__init__(self, name, attributes, **kwargs)
+        super(StructureType, self).__init__(name, attributes, **kwargs)
 
-        # emulate a simple ordered dict
-        self._keys = []
-        self._dict = {}
+        # allow some keys to be hidden:
+        self._visible_keys = []
```

Comment:
Based on reading through what this class does, I don't think you actually need a `_visible_keys` attribute. Instead, you can simply remove or add items from `self._dict`, which is now ordered.